### PR TITLE
Consolidate filter E2E reproductions for the release-x.50.x branch specifically

### PR DESCRIPTION
**Pull request by nemanjaglumac**
This PR mimics #43328, but this time for the `release-x.50.x.` branch specifically.
There should be no notable differences and all tests should pass in CI.

The only difference is that I included repro for #36508 that wasn't originally backported to this branch.

### Additional notes:
- Had to adjust some tests because they were adjusted in `master` to work with the new product features, targeted for 51
- Cherry-picked fea908d474dcedce201c2c22f5888bd4e132e68c, which is #44748

### CODE REVIEW

File: e2e/test/scenarios/filters-reproductions/dashboard-filters-reproductions.cy.spec.js [hunk 1]
```diff
@@ -179,7 +178,7 @@ describe("issue 8030 + 32444", () => {
             cy.findByText(filterDetails.name).click();
             popover().within(() => {
               // the filter is connected only to the first card
-              multiAutocompleteInput().type("1{enter}");
+              cy.get("input").type("1{enter}");
```
Comment (nemanjaglumac) [inline e2e/test/scenarios/filters-reproductions/dashboard-filters-reproductions.cy.spec.js:181]: Reverts this
https://github.com/metabase/metabase/pull/43293/files#diff-9db584711720ed7f09df046944b2c4e4a71ba62532dccb03c89930cffaa9e779L180-R181

File: e2e/test/scenarios/filters-reproductions/dashboard-filters-reproductions.cy.spec.js [hunk 2]
```diff
@@ -548,7 +547,6 @@ describe("issues 15119 and 16112", () => {
     // eslint-disable-next-line no-unscoped-text-selectors -- deprecated usage
     cy.findByText(reviewerFilter.name).click();
     popover().contains("adam").click();
-    multiAutocompleteInput().blur();
```
Comment (nemanjaglumac) [inline e2e/test/scenarios/filters-reproductions/dashboard-filters-reproductions.cy.spec.js]: Reverts this
https://github.com/metabase/metabase/pull/43293/files#diff-9db584711720ed7f09df046944b2c4e4a71ba62532dccb03c89930cffaa9e779R550

File: e2e/test/scenarios/filters-reproductions/filters-reproductions.cy.spec.js [hunk 3]
```diff
@@ -269,7 +269,7 @@ describe("issue 20683", { tags: "@external" }, () => {
     popover().within(() => {
       cy.findByText("Created At").click();
       cy.findByText("Relative dates…").click();
-      cy.findByText("Previous").click();
+      cy.findByText("Past").click();
```
Comment (nemanjaglumac) [inline e2e/test/scenarios/filters-reproductions/filters-reproductions.cy.spec.js:272]: Reverts this
https://github.com/metabase/metabase/pull/44299/files#diff-b8ac0b262f21ddaf28d49d07c061d8919f0013c6980022cafd995aff5b43aa77R272

File: e2e/test/scenarios/filters-reproductions/filters-reproductions.cy.spec.js [hunk 4]
```diff
@@ -538,10 +538,9 @@ describe("issue 25378", () => {
     });
     cy.findByRole("listbox").findByText("months").click();
 
-    popover().within(() => {
-      cy.findByLabelText("Starting from…").click();
-      cy.button("Add filter").click();
-    });
+    popover().findByLabelText("Options").click();
+    popover().last().findByText("Starting from…").click();
+    popover().button("Add filter").click();
```
Comment (nemanjaglumac) [inline e2e/test/scenarios/filters-reproductions/filters-reproductions.cy.spec.js:543]: Reverts this
https://github.com/metabase/metabase/pull/44279/files#diff-b8ac0b262f21ddaf28d49d07c061d8919f0013c6980022cafd995aff5b43aa77L540

**Review by uladzimirdev — APPROVED** [review]

**Pull request merged**